### ISSUE TITLE
RCS1075 should ignore catch blocks with comments

### ISSUE THREAD

**Issue by 0xC0000054** [JosefPihrt/Roslynator#386]
Some .NET Framework APIs are only documented as throwing an Exception.
In those cases RCS1075 flags an empty catch block even if the developer has added a comment describing why the base Exception class is caught and ignored.

One example of this is the [SetResolution](https://msdn.microsoft.com/en-us/library/system.drawing.bitmap.setresolution(v=vs.110).aspx) method in the System.Drawing.Bitmap class.

``` C#
using (Bitmap temp = new Bitmap(32, 32))
{
     try
     {
         temp.SetResolution(96.0F, 96.0F);
     }
     catch (Exception)
     {
          // Ignore any errors when setting the resolution.
     }
}
```

**Comment by Geneticus** [comment]
I would argue that there is no good alternative when programming against any black box API that doesn't document its exceptions. In other cases, your block may yield an exception, but you don't care and it doesn't effect the application by knowing what it was or writing excessive handling code for something you never intend to alert the user about or log..

**Comment by JosefPihrt** [comment]
Rather than suppress a diagnostic when catch block contains a comment I would suggest to supress it using either `#pragma warning`:

```csharp
using (Bitmap temp = new Bitmap(32, 32))
{
    try
    {
        temp.SetResolution(96.0F, 96.0F);
    }
#pragma warning disable RCS1075 // Ignore any errors when setting the resolution.
    catch (Exception)
    {
    }
#pragma warning restore RCS1075
}
```
or `SuppressMessageAttribute`:
```csharp
[SuppressMessage("Design", "RCS1075", Justification = "Ignore any errors when setting the resolution.")]
void M()
{
    using (Bitmap temp = new Bitmap(32, 32))
    {
        try
        {
            temp.SetResolution(96.0F, 96.0F);
        }
        catch (Exception)
        {
        }
    }
}
```

Visual Studio provides UI for this task:

![suppressinsource](https://user-images.githubusercontent.com/11931318/41006668-18f3cab4-6912-11e8-814c-0857553b9d41.png)
